subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -33,22 +33,43 @@ static char *scene_path(char *buf, uint16_t scene, bool vnd, uint8_t page)
 	return buf;
 }
 
+static uint16_t current_scene(const struct bt_mesh_scene_srv *srv, int64_t now)
+{
+	if (model_transition_is_active(&srv->transition) &&
+	    now < srv->transition_end) {
+		return srv->prev;
+	}
+
+	return srv->next;
+}
+
+static uint16_t target_scene(const struct bt_mesh_scene_srv *srv, int64_t now)
+{
+	if (model_transition_is_active(&srv->transition) &&
+	    now < srv->transition_end) {
+		return srv->next;
+	}
+
+	return BT_MESH_SCENE_NONE;
+}
+
 static void scene_status_encode(struct bt_mesh_scene_srv *srv,
 				struct net_buf_simple *buf,
 				enum bt_mesh_scene_status status)
 {
 	int64_t now = k_uptime_get();
+	uint16_t current = current_scene(srv, now);
+	uint16_t target = target_scene(srv, now);
 
 	bt_mesh_model_msg_init(buf, BT_MESH_SCENE_OP_STATUS);
 	net_buf_simple_add_u8(buf, status);
-
-	if (now < srv->transition_end) {
-		net_buf_simple_add_le16(buf, srv->prev);
-		net_buf_simple_add_le16(buf, srv->curr);
+	if (current != target && target != BT_MESH_SCENE_NONE) {
```

Comment:
The current != target is necessary, because if current scene is x and the user set the scene again to x. Then current and target is equal and target is not equal to NONE. Then we do not wan't it to send target state in status.